gulpfile.js

```diff
@@ -20,30 +14,33 @@ const rollupCommon = require('@rollup/plugin-commonjs');
 const rollupReplace = require('@rollup/plugin-replace');
 const { terser: rollupTerser } = require('rollup-plugin-terser');
 
+const readJSON = async path => {
+  const content = await fs.readFile(path, 'utf-8');
+  return JSON.parse(content);
+};
+
 function css() {
   const boundSass = gulpSass(sass);
-  return gulp.src('src/css/*.scss', { sourcemaps: true })
-    .pipe(boundSass.sync().on('error', boundSass.logError))
+  return gulp.src('src/css/*.scss')
     .pipe(gulpSourcemaps.init())
-    .pipe(boundSass({ outputStyle: 'compressed' }))
+    .pipe(boundSass.sync({ outputStyle: 'compressed' }).on('error', boundSass.logError))
     .pipe(gulpSourcemaps.write('./'))
     .pipe(gulp.dest('build/'));
 }
 
-
 async function html() {
   const [config, changelog, headCSS] = await Promise.all([
-    readJSON(`${__dirname}/src/config.json`),
-    readJSON(`${__dirname}/src/changelog.json`),
-    fs.readFile(`${__dirname}/build/head.css`)
+    readJSON(path.join(__dirname, 'src/config.json')),
```

Comment:
Should these use `path.join` for all parts of the path? As in `path.join(__dirname, 'src', 'config.json')`?

Reply:
It's the same and personally I prefer the shorter way :)

Reply:
I'm missing something here… I thought the purpose of `path.join` was to cater for platform differences.

If platform differences don't matter, then `__dirname + '/src/config.json'` is fine.

If platform differences matter, then you'd need `path.join(__dirname, 'src', 'config.json')` surely?

I can't figure out where `path.join(__dirname, 'src/config.json')` fits in this.

Reply:
Because `path.join(__dirname, 'src/config.json')` already fixes any platform differences regardless of the number of parts :)

Reply:
```js
path.join('c:\\foo\\bar', 'src/config.json');
```

…is `c:\foo\bar/src/config.json`, creating a mix of path separators.

Reply:
```js
const path = require('path');

let p = __dirname + '/src/config.json';
console.log(`p: ${p}`); // prints: p: C:\Users\xmr\Desktop/src/config.json

p = path.join(__dirname, 'src', 'config.json');
console.log(`p: ${p}`); // prints: p: C:\Users\xmr\Desktop\src\config.json

p = path.join(__dirname, 'src/config.json');
console.log(`p: ${p}`); // prints: p: C:\Users\xmr\Desktop\src\config.json
```

Reply:
```js
console.log(path.join('c:\\foo\\bar', 'src/config.json')); // prints: c:\foo\bar\src\config.json
```

Reply:
Ah I see, it's because it normalises the path afterwards, and on Windows that means swapping `/` for `\`. On Posix:

```js
console.log(path.join('c:\\foo\\bar', 'src/config.json')); // prints: c:\foo\bar/src/config.json
```

I realise it works because the above will never happen, as the first half will always be Posix on Posix systems, but it feels like a bit of a hack. I'd prefer to use `path.join` for all parts of the path, except the parts within `__dirname` of course as we already know they're in the platform's format.

Reply:
Well, that's because on Posix systems `\` is allowed to be in paths, i.e. is not a separator.

I think we are fine here with the patch as is, but if you insist I can change it, although I don't think it's needed.

Reply:
My problem with it is in works in this one situation, but it doesn't work as a general rule, and requires particular knowledge to know why it's ok. I'd rather go with `path.join(__dirname, 'src', 'config.json')` so it isn't confusing in future.

@TrySound I think we need a tie-break opinion here.

Reply:
I use your solution, either way it's better